packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -82,9 +81,23 @@ export const MenuButton = compose<MenuButtonType>({
         <Slots.button />
         {context.showContextualMenu && (
           <Slots.contextualMenu>
-            {menuItems.map((menuItem) => (
-              <ContextualMenuItem key={menuItem.itemKey} {...menuItem} />
-            ))}
+            {menuItems.map((menuItem) => {
+              const { hasSubmenu, submenu, showSubmenu, componentRef, submenuItems, ...items } = menuItem;
```

Comment:
Instead of having the client pass in the showSubmenu, onHoverIn, and setShowMenu, it would simplify the API even more for them if we defined those callback hooks in here, and pass them in in this file.

Reply:
Good idea, will do

Reply:
@lenahong , what do you think, maybe we can have both options, to have opportunity for customers to use their own method and to have default ones if customer didn't add any?

Reply:
I added both opportunities